infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -396,9 +419,28 @@ private void stopLogReplication() {
     public void processLockAcquire() {
         log.debug("Process lock acquire event");
         isLeader = true;
+
+        // Regardless of the role of this node, on leadership acquisition
+        // we shall query the topology provider to get the most up to date
+        // topologyConfigId, which might have changed during our time as non-leaders
+        // and for which we might have missed the change notification and update it on
+        // the sink manager so the value is cached to filter messages upon receive.
+        fetchTopologyFromClusterManager();
```

Comment:
I think we missed 'logReplicationServerHandler.setActive' after updating localTopology.

Reply:
you're right! So, to avoid this being lose and in different places, I refactored a bit as for updates to always happen when verifying if the cluster belongs to the retrieved topology.